src/components/PostDetails.tsx

```diff
@@ -1,106 +1,139 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { addComment, deleteComment, getCommentsByPost } from '../api/comments';
+
+type Props = {
+  post: Post | null;
```

Comment:
Make sure you described objects in propTypes. In this case, you should describe the `post` prop.